marklogic-data-hub-spark-connector/src/main/java/com/marklogic/hub/spark/sql/sources/v2/reader/HubDataSourceReader.java

```diff
@@ -141,5 +174,49 @@ private int determineNumPartitions(Map<String, String> options) {
     public StructType readSchema() {
         return sparkSchema;
     }
+
+    private JsonNode determineReadRecordsEndpointParams(Map<String, String> options) {
+        ObjectNode endpointParams;
+        if (options.containsKey("readrowsendpointparams")) {
+            try {
+                endpointParams = (ObjectNode) objectMapper.readTree(options.get("readrowsendpointparams"));
+            } catch (IOException e) {
+                throw new IllegalArgumentException("Unable to parse readrowsendpointparams, cause: " + e.getMessage(), e);
+            }
+        } else {
+            endpointParams = objectMapper.createObjectNode();
+        }
+
+        boolean doesNotHaveApiPath = (!endpointParams.hasNonNull("apiPath") || StringUtils.isEmpty(endpointParams.get("apiPath").asText()));
+        boolean hasEndpointConstants = endpointParams.hasNonNull("endpointConstants") && !StringUtils.isEmpty(endpointParams.get("endpointConstants").asText());
+        boolean hasEndpointState = endpointParams.hasNonNull("endpointState") && !StringUtils.isEmpty(endpointParams.get("endpointState").asText());
+        if (doesNotHaveApiPath && hasEndpointState) {
+            throw new IllegalArgumentException("Cannot set endpointState in readrowsendpointparams unless apiPath is defined as well.");
```

Comment:
Good error here, but add "option" after "readrowsendpointparams" to give it a little more context